### PR TITLE
Disallow setting clustering table feature through tblproperties

### PR DESCRIPTION
**Pull request by ami7o**
#### Which Delta project/connector is this regarding?

- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description

It throws an error when we do set tblproperties (delta.feature.clustering="supported") through CREATE TABLE or ALTER TABLE

## How was this patch tested?

existing UTs

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

**Review by imback82 — APPROVED** [review]
LGTM!

**Review by imback82 — COMMENTED** [review]
Actually can we add tests?

**Comment by ami7o** [comment]
This depends on https://github.com/delta-io/delta/pull/2566 to pass the tests.

File: spark/src/main/resources/error/delta-error-classes.json [hunk 1]
```diff
@@ -71,6 +71,12 @@
     ],
     "sqlState" : "42837"
   },
+  "DELTA_ALTER_TABLE_SET_CLUSTERING_TABLE_FEATURE_NOT_ALLOWED" : {
+    "message" : [
+      "Cannot enable <tableFeature> table feature using SET TBLPROPERTIES. Please use CREATE TABLE CLUSTER BY to create a Delta table with clustering."
```
Comment (imback82) [inline spark/src/main/resources/error/delta-error-classes.json:76]:
```suggestion
      "Cannot enable <tableFeature> table feature using ALTER TABLE SET TBLPROPERTIES. Please use CREATE or REPLACE TABLE CLUSTER BY to create a Delta table with clustering."
```

File: spark/src/test/scala/org/apache/spark/sql/delta/clustering/ClusteringTableFeatureSuite.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    withTable("tbl") {
+      val e = intercept[DeltaAnalysisException] {
+        sql("CREATE TABLE tbl(a INT, b STRING) USING DELTA " +
+          "TBLPROPERTIES('delta.feature.clustering' = 'supported')")
+      }
+      checkError(
+        e,
+        "DELTA_CREATE_TABLE_SET_CLUSTERING_TABLE_FEATURE_NOT_ALLOWED",
+        parameters = Map("tableFeature" -> "clustering")
+      )
```
Comment (imback82) [inline spark/src/test/scala/org/apache/spark/sql/delta/clustering/ClusteringTableFeatureSuite.scala:36]: To be consistent with `sql`:
```suggestion
        parameters = Map("tableFeature" -> "clustering"))
```

File: spark/src/test/scala/org/apache/spark/sql/delta/clustering/ClusteringTableFeatureSuite.scala [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    withTable("tbl") {
+      sql("CREATE TABLE tbl(a INT, b STRING) USING DELTA")
+      val e = intercept[DeltaAnalysisException] {
+        sql("ALTER TABLE tbl SET TBLPROPERTIES ('delta.feature.clustering' = 'supported')")
+      }
+      checkError(
+        e,
+        "DELTA_ALTER_TABLE_SET_CLUSTERING_TABLE_FEATURE_NOT_ALLOWED",
+        parameters = Map("tableFeature" -> "clustering")
+      )
```
Comment (imback82) [inline spark/src/test/scala/org/apache/spark/sql/delta/clustering/ClusteringTableFeatureSuite.scala:50]: ditto

**Review by imback82 — APPROVED** [review]
LGTM!

**Pull request closed without merging**